server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java

```diff
@@ -0,0 +1,142 @@
+  /** 
+   * The error string sent by the response object in doPost when the body of the 
+   * POST request does not have a "userId" field in its "person" field.
+   */
+  public static final String NO_USER_ID_ERROR = "No \"userId\" found in JSON.";
+  /** Message to be logged when the body of the POST request does not have required fields. */
+  public static final String LOG_INPUT_ERROR_MESSAGE =
+      "Error with JSON input in UpdatePersonServlet: ";
+  /** Name of the key in the input JSON that corresponds to the person object. */
+  public static final String PERSON_FIELD_NAME = "person";
+  /** Name of the key in the input JSON that corresponds to the update mask. */
+  public static final String UPDATE_MASK_FIELD_NAME = "updateMask";
+
+  private static final Gson gson = new Gson();
+  private static StorageHandlerApi storageHandler;
+
+  /** 
+   * Overloaded constructor for dependency injection.
+   * @param storageHandler the {@link StorageHandlerApi} that is used when fetching the Person
+   */
+  public UpdatePersonServlet(StorageHandlerApi storageHandler) {
+    super();
+    this.storageHandler = storageHandler;
+  }
+
+  /** 
+   * Explicit default constructor used for instantiating the servlet when not testing.
+   */
+  public UpdatePersonServlet() {
+    super();
+    this.storageHandler = new StorageHandlerApi();
+  }
+
+  /** 
+   * Updates a Person object in the database and returns that object in JSON format.
+   * @param request the POST request that must have a valid JSON representation of the Person to be
+   *     updated as well as an optional mask of fields that will be updated as its body. The
+   *     Person to be updated must be represnted by a {@code "person"} key, who's value is an 
+   *     object that represents the updated Person. The update mask must have the key
+   *     {@code "updateMask"} and the value of a comma separated list of the fields to be updated
+   *     from the "person" key. If no update mask exists, all fields from the "person" key will be
+   *     used to update the Person. If the request does not have either of these keys, or is
+   *     syntactically incorrect, the response object will send a "400 Bad Request error"
+   * @param response the response from this method, will contain the updated Person in JSON format.
+   *     If the request object does not have a valid JSON body (as described in the request
+   *     parameter) this object will send a "400 Bad Request error"
+   * @throws IOException if an input or output error is detected when the servlet handles the request
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Person personToUpdate;
+    try {
+      // Get the update mask / updated Person from the post request.
+      Map mappedRequest = gson.fromJson(request.getReader(), Map.class);
+      Map personInfo = (Map) mappedRequest.getOrDefault(PERSON_FIELD_NAME, null);
```

Comment:
nit: Technically this is equivalent to just calling `.get`, which defaults to null

Reply:
Haha I realized this too late. All of the codebase uses getOrDefault() so I figured I'd stay consistent even if it's a little redundant.

Reply:
Ok SG